tools/src/main/scala/scala/scalanative/build/Config.scala

```diff
@@ -94,50 +145,54 @@ sealed trait Config {
     }
   }
 
-  private[scalanative] lazy val targetsMac = Platform.isMac ||
+  private[scalanative] lazy val targetsMac: Boolean = Platform.isMac ||
     compilerConfig.targetTriple.exists { customTriple =>
       Seq("mac", "apple", "darwin").exists(customTriple.contains(_))
     }
 }
 
+/** Factory to create [[#empty]] [[Config]] objects */
 object Config {
 
-  /** Default empty config object where all of the fields are left blank. */
+  /** Default empty config object where all of the fields are left blank or the
+   *  default value.
+   */
   def empty: Config =
     Impl(
-      nativelib = Paths.get(""),
-      mainClass = None,
-      classPath = Seq.empty,
       basedir = Paths.get(""),
       testConfig = false,
+      moduleName = "",
+      mainClass = None,
+      classPath = Seq.empty,
       logger = Logger.default,
       compilerConfig = NativeConfig.empty
     )
 
   private final case class Impl(
-      nativelib: Path,
-      mainClass: Option[String],
-      classPath: Seq[Path],
       basedir: Path,
       testConfig: Boolean,
+      moduleName: String,
+      mainClass: Option[String],
+      classPath: Seq[Path],
       logger: Logger,
       compilerConfig: NativeConfig
   ) extends Config {
-    def withNativelib(value: Path): Config =
-      copy(nativelib = value)
```

Comment:
I wondered if the removal of it can have a negative impact on build tools integration. 
From the mill sources, I can see that it was never used for `0.4.x` versions ([first version of its integration](https://github.com/com-lihaoyi/mill/blob/82ea8738ef31f082a4afc2b553b2bee7a550b267/scalanativelib/worker/0.4/src/ScalaNativeWorkerImpl.scala)) so there should no problems with future upgrades.

Reply:
It looks like they have versioned `0.4` in the path so maybe they can do `0.5` next, hopefully.